Pr udi update repo comment seperate comments

udiraz2 opened the pull request:
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](https://github.com/checkmarx-ts/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/checkmarx-ltd/open-source-template/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> CxFlow used to add comments to PR every time a scan was initiated for the PR. This caused a lot of verbosity in the PR page. Now CxFlow updates existing comments to reduce verbosity.

### References

> https://github.com/checkmarx-ltd/cx-flow/issues/185

### Testing

> There are two automation (integration) tests. First tests a PR without any comments and validates that there are two CxFlow comments. Second test tests for a PR that already have comments, and validates that one comment is updated.
> To test manually, you should add a webhook that points t CxFlow machine. open a pull request, and CxFlow should scan and create/update the comments in the PR

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentaiton is a Wiki Update, please indicate desired changes within PR MD Comment*
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

cxflowtestuser commented:
Scan submitted to Checkmarx

cxflowtestuser commented:
### Checkmarx scan completed
[Full Scan Details](http://EC2AMAZ-JP2I9QF/CxWebClient/ViewerMain.aspx?scanid=1000145&projectid=59)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|549
Low|5771
Informational|44

#### Violation Summary
Severity|Count
---|---

#### Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---

udiraz2 commented:
Fixed update comments for pull request logic, and added tests.
Now CxFlow updates PR comments if necessary, and does not create new comments for every scan.
